blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineStepVisitor.java

```diff
@@ -104,8 +103,13 @@ public void chunkEnd(@Nonnull FlowNode endNode, @CheckForNull FlowNode afterChun
         super.chunkEnd(endNode, afterChunk, scanner);
         if(endNode instanceof StepEndNode && PipelineNodeUtil.isStage(((StepEndNode)endNode).getStartNode())){
             currentStage = ((StepEndNode)endNode).getStartNode();
+        } else {
```

Comment:
Yikes, is there a better way we can deal with random pipeline events? e.g. just reload the whole pipeline on the front end?

Reply:
Why you think this change is bad? All we are doing is finding enclosing block, upstream library has it parsed/cached optimally. 

Reloading the pipeline will execute this very code and we won’t return the right steps.